java/clients/src/main/java/sleeper/clients/QueryWebSocketClient.java

```diff
@@ -245,6 +276,7 @@ public void onMessage(String json) {
                             .forEach(out::println);
                 }
                 queryComplete = true;
+                future.complete(getResults(currentQueryId));
```

Comment:
It looks like this will still happen if there was an exception earlier, since it's just calling completeExceptionally instead of throwing an exception. What should happen? Why isn't this failing a unit test?

Reply:
Ah ok, calling complete doesn't do anything if it's already completed exceptionally. Thanks.